SgmlTests/Tests-Logic.cs

```diff
@@ -9,143 +9,168 @@
-                    new DesktopEntityResolver());
-                dtd.Name = docType;
-                return dtd;
+                if (stream is null) throw new ArgumentException("unable to load requested resource: " + name);
+                using (StreamReader sr = new StreamReader(stream))
+                {
+                    return sr.ReadToEnd();
+                }
             }
         }
 
-        private static string RunTest(CaseFolding caseFolding, string doctype, bool format, string source, XmlReaderTestCallback callback) {
+        /// <summary></summary>
+        /// <param name="docType"></param>
+        /// <param name="name">The name of the EmbeddedResource file located under the <c>SgmlTests\Resources</c> project directory.</param>
+        /// <returns></returns>
+        internal static SgmlDtd LoadDtd(string docType, string resourceFileName)
+        {
+            StringReader dtdTextReader = new StringReader(ReadTestResource(resourceFileName));
+
+            SgmlDtd dtd = SgmlDtd.Parse(baseUri: null, name: docType, input: dtdTextReader, subset: "", nt: new NameTable(), resolver: new DesktopEntityResolver());
+            return dtd;
+        }
+
+        private static string RunTest(CaseFolding caseFolding, string doctype, bool format, string source, XmlReaderTestCallback callback)
+        {
 
             // initialize sgml reader
-            XmlReader reader = new SgmlReader {
+            SgmlReader sgmlReader = new SgmlReader
+            {
                 CaseFolding = caseFolding,
                 DocType = doctype,
                 InputStream = new StringReader(source),
                 WhitespaceHandling = format ? WhitespaceHandling.None : WhitespaceHandling.All
             };
+
             if (doctype == "OFX")
             {
-                ((SgmlReader)reader).Dtd = LoadDtd("OFX", "ofx160.dtd");
+                sgmlReader.Dtd = LoadDtd("OFX", "ofx160.dtd");
             }
 
-
             // check if we need to use the LoggingXmlReader
-            if (_debug) {
-                reader = new LoggingXmlReader(reader, Console.Out);
+            XmlReader reader = sgmlReader;
+#if DEBUG
+            {
+                reader = new LoggingXmlReader(sgmlReader, Console.Out);
             }
+#endif
 
             // initialize xml writer
-            var stringWriter = new StringWriter();
-            var xmlTextWriter = new XmlTextWriter(stringWriter);
-            if(format) {
+            StringWriter stringWriter = new StringWriter();
+            XmlTextWriter xmlTextWriter = new XmlTextWriter(stringWriter);
+            if (format)
+            {
                 xmlTextWriter.Formatting = Formatting.Indented;
             }
             callback(reader, xmlTextWriter);
             xmlTextWriter.Close();
 
             // reproduce the parsed document
-            var actual = stringWriter.ToString();
+            string actual = stringWriter.ToString();
```

Comment:
why remove "var" here?

Reply:
Auto-formatting made that change.

Reply:
ok, I see you changed your auto-formatter to move it back to var, thanks.  What editor are you using anyway that does this kind of auto-rewriting during formatting?

Reply:
[VS2019's _Code Cleanup_ utility under the Analyze menu](https://docs.microsoft.com/en-us/visualstudio/ide/code-styles-and-code-cleanup?view=vs-2019). It doesn't respect `.editorconfig` yet, unfortunately. Most of the time I just use the _Format Document_ command which does respect `.editorconfig` but it only supports a subset of `.editorconfig` features and doesn't perform more aggressive changes.